File: app/src/main/java/com/ppy/nfcsample/card/reader/CardReaderChain.java
package com.ppy.nfcsample.card.reader;

import com.ppy.nfclib.NfcCardReaderManager;
import com.ppy.nfcsample.card.DefaultCardInfo;

import java.io.IOException;
import java.util.List;

/**
 * Created by ZP on 2018/1/4.
 */

public class CardReaderChain implements IReader.Chain {

    private List<IReader> mReaders;

    public CardReaderChain(List<IReader> readers) {
        mReaders = readers;
    }

    @Override
    public DefaultCardInfo proceed(NfcCardReaderManager nfcCardReaderManager) throws IOException {
        DefaultCardInfo defaultCardInfo = null;
        for (IReader reader : mReaders) {
            defaultCardInfo = reader.readCard(nfcCardReaderManager);
            if (defaultCardInfo != null) {
                break;
            }
        }
        return defaultCardInfo;
    }
}

File: app/src/main/java/com/ppy/nfcsample/card/reader/CardReaderChain.kt
package com.ppy.nfcsample.card.reader

import com.ppy.nfclib.NfcManagerCompat
import com.ppy.nfcsample.card.DefaultCardInfo

import java.io.IOException

/**
 * Created by ZP on 2018/1/4.
 */

class CardReaderChain(private val mReaders: List<IReader>) : IReader.Chain {

    @Throws(IOException::class)
    override fun proceed(nfcCardReaderManager: NfcManagerCompat): DefaultCardInfo? {
        var defaultCardInfo: DefaultCardInfo? = null
        for (reader in mReaders) {
            defaultCardInfo = reader.readCard(nfcCardReaderManager)
            if (defaultCardInfo != null) {
                break
            }
        }
        return defaultCardInfo
    }
}
